src/schemas/tokens-info.ts

```diff
@@ -26,34 +26,61 @@ const checkMinVersion: CustomValidator = (value) => {
   }
 }
 
-const TokensInfoSchema = Joi.object().pattern(
-  AddressSchema,
+const BaseTokenInfoSchema = Joi.object({
+  address: AddressSchema,
+  imageUrl: Joi.string()
+    // For now only allow assets within this repo
+    .pattern(
+      /^https:\/\/raw.githubusercontent.com\/valora-inc\/address-metadata\/main\/assets\/tokens\/[^/]+\.png$/,
+    )
+    .uri()
+    .custom(checkMatchingAsset, 'has a matching asset'),
+  name: Joi.string().required(),
+  decimals: Joi.number().required(),
+  symbol: Joi.string().required(),
+  isCoreToken: Joi.boolean(),
+  isSupercharged: Joi.boolean(),
+  pegTo: AddressSchema,
+  // Forcing flag to be false if decimals is not 18
+  isSwappable: Joi.when('decimals', {
+    is: 18,
+    then: Joi.boolean(),
+    otherwise: Joi.valid(false),
+  }),
+  minimumAppVersionToSwap: Joi.string()
+    .pattern(/^\d+\.\d+\.\d+$/)
+    .custom(checkMinVersion, 'has a valid version'),
+  isNative: Joi.boolean(),
+})
+
+export const TokenInfoSchema = Joi.alternatives().try(
+  Joi.object({
+    // native tokens don't have an address except CELO
+    isNative: Joi.valid(true).required(),
+    symbol: Joi.string().invalid('CELO').required(),
```

Comment:
We will need to update this value if we add a chain with a native token with an address, right?
Maybe we can use a list here and use the list in the other case too, so we only need to add the symbol to the list.

Reply:
I also wonder if this check is necessary. I see the point of it, but maybe we can just simplify the logic to allow an optional `address` if the token is marked as native?

This check sounds like extra logic specific to how Celo works.

Reply:
I've never heard of another blockchain with a native token that has an address. But if we did find one, it could just be added here

Reply:
I'm inclined to keep the check as-is, since it gives us stricter guarantees on data quality